core/src/test/java/com/linecorp/armeria/client/HttpClientAdditionalHeadersTest.java

```diff
@@ -50,10 +53,24 @@ void disallowedHeadersMustBeFiltered() {
                          })
                          .build();
 
-        assertThat(client.get("/").aggregate().join().contentUtf8())
-                .doesNotContain("=https")
-                .doesNotContain("=503")
-                .doesNotContain("=CONNECT")
-                .contains("foo=bar");
+        try (ClientRequestContextCaptor clientCaptor = Clients.newContextCaptor()) {
+            assertThat(client.get("/").aggregate().join().contentUtf8())
+                    .doesNotContain("=https")
+                    .doesNotContain("=503")
+                    .doesNotContain("=CONNECT")
+                    .contains("foo=bar");
+            assertThat(clientCaptor.size()).isEqualTo(1);
+            final ClientRequestContext clientContext = clientCaptor.get();
+            assertThat(clientContext.authority()).isEqualTo(server.httpEndpoint().authority());
+            assertThat(clientContext.log().ensureComplete().requestHeaders().get(HttpHeaderNames.USER_AGENT))
+                    .isEqualTo(UserAgentUtil.USER_AGENT.toString());
+
+            final ServiceRequestContextCaptor serviceCaptor = server.requestContextCaptor();
+            assertThat(serviceCaptor.size()).isEqualTo(1);
+            final ServiceRequestContext serviceContext = serviceCaptor.poll();
+            assertThat(serviceContext.request().authority()).isEqualTo(server.httpEndpoint().authority());
+            assertThat(serviceContext.request().headers().get(HttpHeaderNames.USER_AGENT))
+                    .isEqualTo(UserAgentUtil.USER_AGENT.toString());
+        }
```

Comment:
How about moving this check into a separate test method and name it properly? The method name `disallowedHeadersMustBeFiltered()` doesn't sound relevant to checking the `USER_AGENT` header.

Reply:
Good point 👍  Done and added a couple test cases which covers all scenarios: internal, default, request, additional headers